src/test/java/org/zalando/planb/provider/RealmConfigTest.java

```diff
@@ -16,4 +20,28 @@ public void testEnsureLeadingSlashWhenMissing() throws Exception {
     public void testEnsureLeadingSlashWhenPresent() throws Exception {
         assertThat(ensureLeadingSlash("/foo")).isEqualTo("/foo");
     }
+
+    @Test
+    public void testStripLeadingSlash() {
+        assertThat(stripLeadingSlash("foo")).isEqualTo("foo");
+        assertThat(stripLeadingSlash("/foo")).isEqualTo("foo");
+    }
+
+    @Test
+    public void testFindRealmInHostEmpty() {
+        RealmConfig config = new RealmConfig();
+        assertThat(config.findRealmNameInHost("foo")).isEmpty();
+    }
+
+    @Test
+    public void testFindRealmInHost() {
+        Set<String> realmNames = ImmutableSet.of("/foo", "/bar");
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "somethingelse")).isEmpty();
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "foobar")).isEmpty();
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "xfoox")).isEmpty();
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "provider.foo.example.org")).contains("/foo");
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "provider-bar.example.org")).contains("/bar");
+        // we sort the realm names, so it will return the first one if both match
+        assertThat(RealmConfig.findRealmNameInHost(realmNames, "foo.bar")).contains("/bar");
```

Comment:
and maybe add a test if we decide to avoid the null check on the controller side:

```
assertThat(RealmConfig.findRealmNameInHost(realmNames, null)) -> throw null pointer
```